scRNA-seq/04-dimension_reduction_scRNA.Rmd

```diff
@@ -212,13 +212,53 @@ dim(filtered_sce)
 Now we will perform the same normalization steps we did in a previous dataset, using `scran::computeSumFactors()` and `scater::logNormCounts()`.
 You might recall that there is a bit of randomness in some of these calculations, so we should be sure to have used `set.seed()` earlier in the notebook for reproducibility.
 
-```{r normalize}
+```{r sumfactors}
 # Cluster similar cells
 qclust <- scran::quickCluster(filtered_sce)
 
 # Compute sum factors for each cell cluster grouping.
+filtered_sce <- scran::computeSumFactors(filtered_sce, clusters = qclust, positive = FALSE)
+```
+
+It turns out in this case we end up with some negative size factors.
+This is usually an indication that our filtering was not stringent enough, and there remain a number of cells or genes with nearly zero counts.
+This probably happened when some cells had many UMIs from the genes we removed in the last filtering.
```

Comment:
The wording of this sentence could use another pass, but not sure what to suggest, since I don't quite follow it.

Reply:
How about this?
```suggestion
This probably happened when we removed the infrequently-expressed genes; cells which had high counts from those particular genes (and few others) could have had their total counts dramatically reduced.
```